### PR TITLE
auth: only set x-auth-project-id header if truthy

### PR DESCRIPTION
**Pull request by evanlucas**
This makes all tests pass on iojs.
https://github.com/iojs/io.js/commit/979d0ca874df0383311ca06f154f6965074
196ee introduced throwing an error when the header value is undefined.

### CODE REVIEW

**Comment by kenperkins** [comment]
Should this maybe accompany a change to package.json regarding `engine` ?

File: lib/pkgcloud/common/auth.js [hunk 1]
```diff
@@ -24,7 +24,9 @@ auth.basic = function basicAuth(req) {
 // Add Account number for requests to rackspace API
 auth.accountId = function (req) {
   req.headers = req.headers || {};
-  req.headers['x-auth-project-id'] = this.config.accountNumber;
+  if (this.config.accountNumber) {
+    req.headers['x-auth-project-id'] = this.config.accountNumber;
+  }
```
Comment (kenperkins) [inline lib/pkgcloud/common/auth.js]: I have a feeling this code path isn't even used anywhere anymore. Although I can't recall for certain.

**Comment by evanlucas** [comment]
`npm` checks the engine in https://github.com/npm/npm/blob/master/lib/cache/add-named.js#L87-L103. When `npm` sets the `node-version` config flag, it is based on `process.version`. So even with `iojs`, if `engine-strict` is enabled, it will still be satisfied by `iojs`. 

```
var semver = require('semver')
semver.satisfies('v1.8.1', '>=0.10.x', true) // => true
```

File: lib/pkgcloud/common/auth.js (hunk 1 above)
Comment (evanlucas) [inline lib/pkgcloud/common/auth.js]: It is still used in the following after a quick grep:

```
./pkgcloud/hp/database/client/index.js:  
17:  this.before.push(auth.accountId); 
  
./pkgcloud/openstack/database/client/index.js:  
17:  this.before.push(auth.accountId); 
  
./pkgcloud/rackspace/database/client/index.js:  
17:  this.before.push(auth.accountId); 
```

Comment (kenperkins) [inline lib/pkgcloud/common/auth.js]: Hm. I seem to recall that accountId is a part of an older API specification. This is probably good for now, although the Trove API needs major TLC.

**Comment by kenperkins** [comment]
LGTM.

**Comment by kenperkins** [comment]
Thanks @evanlucas!

**Pull request merged**